modules/gapi/test/common/gapi_core_tests_inl.hpp

```diff
@@ -1267,6 +1267,193 @@ TEST_P(NormalizeTest, Test)
     }
 }
 
+TEST_P(BackendOutputTest, EmptyOutput)
+{
+    // G-API code //////////////////////////////////////////////////////////////
+    cv::GMat in1, in2, out;
+    out = cv::gapi::mul(in1, in2);
+    cv::GComputation c(cv::GIn(in1, in2), cv::GOut(out));
+    c.apply(cv::gin(in_mat1, in_mat2), cv::gout(out_mat_gapi), getCompileArgs());
```

Comment:
Test not only tests but also _illustrates_ (and I believe all the Great tests are mainly illustrations but not dumb mechanical checks).

Put an assert here (before calling apply()) that `out_mat_gapi` is empty.
Put an assert (after calling) that it is not empty anymore.

This is the contract we're actually want to to test and illustrate.

Reply:
added asserts here